lib/usecase/login_use_case.dart

```diff
@@ -0,0 +1,30 @@
+import 'dart:async';
+
+import 'package:flutter_survey/api/exception/network_exceptions.dart';
+import 'package:flutter_survey/api/repository/oauth_repository.dart';
+import 'package:flutter_survey/usecase/base/base_use_case.dart';
+import 'package:injectable/injectable.dart';
+
+class LoginInput {
+  final String email;
+  final String password;
+
+  LoginInput({required this.email, required this.password});
+}
+
+@Injectable()
+class LoginUseCase extends UseCase<void, LoginInput> {
+  final OauthRepository _repository;
+
+  const LoginUseCase(this._repository);
+
+  @override
+  Future<Result<void>> call(LoginInput credential) {
+    return _repository
+        .login(email: credential.email, password: credential.password)
+        .then((value) =>
+            Success(null) as Result<void>) // ignore: unnecessary_cast
+        .onError<Exception>((err, stackTrace) => Failed(
+            UseCaseException(NetworkExceptions.fromDioException(err), err)));
```

Comment:
yep - here is what I mentioned before: a UseCase shouldn't know about Dio by mentioning `fromDioException()` yeah?

Reply:
if it happens to be an error that's not NetworkExceptions, by constructing `UseCaseException(networkExceptions, )` - you may face a runtime casting exception. Better ensure that you throw back the right type from Repository layer 😛